impexp-client/src/main/java/org/citydb/cli/ImpExpCli.java

```diff
@@ -299,4 +311,118 @@ private void initDBPool() throws ImpExpException {
 
 		return files;
 	}
+
+	public boolean calcBoundingBox() throws ImpExpException {
+		initDBPool();
+		if (!dbPool.isConnected())
+			throw new ImpExpException("Connection to database could not be established.");
+
+		log.info("Initializing database export...");
+
+		EventDispatcher eventDispatcher = ObjectRegistry.getInstance().getEventDispatcher();
+		boolean success = false;
+
+		try {
+			BoundingBox bbox = doCalcBBOX();
+			success = (bbox != null);
+		} catch (Exception e) {
+			throw new ImpExpException("BBOX calculation failed due to an internal error.", e);
+		} finally {
+			try {
+				eventDispatcher.flushEvents();
+			} catch (InterruptedException e) {
+
+			}
+			dbPool.disconnect();
+		}
+
+		if (success)
+			log.info("BBOX successfully calculated.");
+		else
+			log.warn("BBOX calculation aborted.");
+
+		return success;
+	}
+
+	private BoundingBox doCalcBBOX() {
+		final ReentrantLock mainLock = new ReentrantLock();
+		final ReentrantLock lock = mainLock;
+		lock.lock();
+
+		BoundingBox bbox = null;
+		DatabaseConnectionPool dbConnectionPool = DatabaseConnectionPool.getInstance();
+		try {
+			log.info("Calculating bounding box...");
+			if (dbConnectionPool.getActiveDatabaseAdapter().hasVersioningSupport())
+				return null;
+
+			try {
+				FeatureType featureType = schemaMapping.getFeatureType(config.getProject().getDatabase().getOperation().getBoundingBoxTypeName());
+				//FeatureType featureType = config.getProject().getKmlExporter().getQuery().getFeatureTypeFilter().
```

Comment:
Maybe the feature type filter shall be taken into account here for calculating the bounding box when running KML-Export?